src/httpserver.cpp

```diff
@@ -477,6 +475,10 @@ void StopHTTPServer()
         event_base_free(eventBase);
         eventBase = nullptr;
     }
+    if (workQueue) {
```

Comment:
I don't think this is the correct fix. By the time `eventHTTP` / `eventBase` is freed it needs to be sure that there are no workers any more, and that there's nothing in the work queue. After all, work in progress would refer to them.

Reply:
In latest commit:
 - prevent adding stuff to queue after interrupt
 - after interrupt queue is drained

So at this point queue is empty and concurrent `http_request_cb` fail to enqueue.